src/store/VideoReducer.js

```diff
@@ -0,0 +1,39 @@
+const VideoReducer = (state, action) => {
+  switch (action.type) {
+    case 'SET_CURRENT_VIDEO':
+      return { ...state, currentVideo: action.payload };
+    case 'ADD_FAVORITE_VIDEO': {
+      const videoExists = () => {
```

Comment:
You don't need to declare these functions inside the case, you can extract them and leave the case cleaner.